From f89998d008cae4f7ed4351204afa297dcb43eb18 Mon Sep 17 00:00:00 2001
From: ctang <ctang.ma@gmail.com>
Date: Thu, 2 Jul 2015 09:56:40 -0400
Subject: [PATCH 0171/1164] HIVE-11157: Hive.get(HiveConf) returns same Hive
 object to different user sessions (Chaoyu Tang,
 reviewed by Jimmy Xiang)

---
 .../org/apache/hadoop/hive/ql/metadata/Hive.java   |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java b/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
index 5cb9c13..fc149cc 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
@@ -185,7 +185,7 @@ public static Hive get(Configuration c, Class<?> clazz) throws HiveException {
   /**
    * Gets hive object for the current thread. If one is not initialized then a
    * new one is created If the new configuration is different in metadata conf
-   * vars then a new one is created.
+   * vars, or the owner will be different then a new one is created.
    *
    * @param c
    *          new Hive Configuration
@@ -195,7 +195,7 @@ public static Hive get(Configuration c, Class<?> clazz) throws HiveException {
    */
   public static Hive get(HiveConf c) throws HiveException {
     Hive db = hiveDB.get();
-    if (db == null ||
+    if (db == null || !db.isCurrentUserOwner() ||
         (db.metaStoreClient != null && !db.metaStoreClient.isCompatibleWith(c))) {
       return get(c, true);
     }
@@ -283,6 +283,9 @@ private void close() {
       metaStoreClient.close();
       metaStoreClient = null;
     }
+    if (owner != null) {
+      owner = null;
+    }
   }
 
   /**
-- 
1.7.9.5

